### PR TITLE
Build multi-arch images, retry mechanism for builds, improved releases

### LINKED ISSUE #2653: Consider switching away from newsfragments
I found `newsfragments` extremely inconvenient.

It is basically impossible to understand what is going on in Mailu in terms of development with reasonable effort when changes are scattered across numerous single-line files named randomly (after PR number). There is basically no way for normal humans to quickly see what happened between 1.9.46 and 1.9.47 exactly. I have never seen such approach in other projects and I'm honestly happy that I didn't.

Moreover, current releases are built from 1.9 branch if I read it correctly, yet release notes recommend to go check `newsfragments` in `master` branch, which is not helpful as it has nothing to do with 1.9 branch even though some fixes are backported (even then, PR numbers of backports are different).

I guess those files are somewhat convenient for machine processing, but consider updating normal changelog file with patch releases for humans too.

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

feature

## What does this PR do?
Switch to building multi-arch images. The images build for pull requests, master and production are now multi-arch images for the architectures:
* linux/amd64
* linux/arm64/v8
* linux/arm/v7

Enhance CI/CD workflow with retry functionality. All steps for building images are now automatically retried. If a build temporarily fails due to a network error, the retried step will still succeed.

Provide a changelog for minor releases. The github release will now:
* Provide the changelog message from the newsfragment of the PR that triggered the backport.
* Provide a github link to the PR/issue of the PR that was backported.

### Related issue(s)
- #1582
- #1200
- #2215
- closes #2653

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by Diman0** [comment]
Currently the tests are not run for the arm images. Should this also be implemented? 
It does mean that the workflow will take a bit longer. However, it does mean we catch any arm-specific issues which prevent the images from starting normally.

**Comment by Diman0** [comment]
bors try

**Comment by Diman0** [comment]
bors try

File: docs/compose/requirements.rst [hunk 1]
```diff
@@ -23,8 +23,8 @@ the latest Linux kernel. The minimal required memory and swap are:
 Pick a distribution
 -------------------
 
-The mail server runs as a set of Docker containers, so it is almost operating 
-system agnostic as long as a fairly recent Linux kernel is running and 
+The mail server runs as a set of Docker containers, so it is almost operating
+system agnostic as long as a fairly recent Linux kernel is running and
```
[the remark this answers is not in the document]
Comment (nextgens) [reply docs/compose/requirements.rst:27]: on windows or elsewhere the linux kernel will be emulated

File: docs/compose/requirements.rst [hunk 2]
```diff
@@ -23,8 +23,8 @@ the latest Linux kernel. The minimal required memory and swap are:
...
 the Docker API (>= 1.11) is available.
```
[the remark this answers is not in the document]
Comment (Diman0) [reply docs/compose/requirements.rst:28]: I'm fine with removing this.

File: docs/compose/requirements.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (nextgens) [reply docs/compose/requirements.rst:27]: It should be all transparent ; see https://docs.docker.com/desktop/windows/wsl/

Not something we need to be explicit about IMHO

File: docs/setup.rst [hunk 3]
```diff
@@ -17,11 +17,12 @@ long-term professional support, you should probably turn to them instead.
...
 
-Mailu images are designed to work on x86 or equivalent hardware, so it
+Mailu ships multi-arch images which are designed to work on linux/amd64,
+linux/arm64v8 or linux/armv7 hardware, so it
 should run on pretty much any cloud server as long as enough power is
-provided. For non x86 machines, see :ref:`arm_images`
+provided.
 
-You are free to choose any operating system that runs Docker (>= 1.11),
+You are free to choose any linux operating system that runs Docker (>= 1.11),
```
Comment (nextgens) [inline docs/setup.rst:25]:
```suggestion
```

**Review by nextgens — CHANGES_REQUESTED** [review]

**Review by nextgens — DISMISSED** [review]

**Pull request merged**